### ISSUE TITLE
Support Parquet read of pages with Encoding::PlainDictionary and non-optional values

### ISSUE THREAD

**Issue by mdrach** [jorgecarleitao/arrow2#668]
I'm reading a parquet file generated by Snowflake and am hitting an error here: https://github.com/jorgecarleitao/arrow2/blob/9d4107ce84d729818ccef2be6bfd979fffc5630c/src/io/parquet/read/fixed_size_binary.rs#L230

```
Decoding "PlainDictionary"-encoded, dictionary-encoded required V1 pages is not yet implemented for FixedSizeBinary
```

Since PlainDictionary encoding with optional values is implemented, would it be simple to implement the non-optional version?
` (Encoding::PlainDictionary, Some(dict), true) => ...`  

(I actually took a stab at this [here](https://github.com/jorgecarleitao/arrow2/compare/main...mdrach:support-dictionary-encoded-nonoptional-pages?expand=1) but am missing a lot of context.)

**Comment by jorgecarleitao** [comment]
Hey. Thanks! Yes, that looks about right. I would just add an integration test against pyarrow here. What I usually do is:

1. add a new column [here](https://github.com/jorgecarleitao/arrow2/blob/main/parquet_integration/write_parquet.py#L52) for the type
2. add the corresponding data [here](https://github.com/jorgecarleitao/arrow2/blob/main/tests/it/io/parquet/mod.rs#L400)
3. add a test like this: https://github.com/jorgecarleitao/arrow2/blob/main/tests/it/io/parquet/read.rs#L139

so that we prove that we can read the type when it is written by pyarrow (which uses the c++ parquet implementation)

**Comment by jorgecarleitao** [comment]
Closed by #683
